akka-projection-testkit/src/main/scala/akka/projection/testkit/scaladsl/ProjectionTestKit.scala

```diff
@@ -63,6 +63,7 @@ final class ProjectionTestKit private[akka] (testKit: ActorTestKit) {
 
   def runWithTestSink[T](projection: Projection[_]): TestSubscriber.Probe[Done] = {
     val sinkProbe = TestSink.probe[Done](testKit.system.toClassic)
+    // FIXME handler.stop is not called when running like this
```

Comment:
With the stop flow you introduced, the handler stop will be called if there is a failure. And that's good and what we expect. 
 
For graceful stop, I think this should be on the user side. This is like a running projection, if you don't stop it, it keeps running.

Reply:
I don't understand what you mean. I find it confusing that the testkit would behave differently for if it's run with this TestSink vs the other run. Anyway, that is for another day.

Reply:
created issue https://github.com/akka/akka-projection/issues/134 for this